### PR TITLE
Remove List.headAndTail and some tuples

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/MethodCalls.fs [hunk 1]
```diff
@@ -1867,12 +1867,15 @@ let GenWitnessExpr amap g m (traitInfo: TraitConstraintInfo) argExprs =
 
         // Fix bug 1281: If we resolve to an instance method on a struct and we haven't yet taken 
         // the address of the object then go do that 
-        if minfo.IsStruct && minfo.IsInstance && (match argExprs with [] -> false | h :: _ -> not (isByrefTy g (tyOfExpr g h))) then 
-            let h, t = List.headAndTail argExprs
-            let wrap, h', _readonly, _writeonly = mkExprAddrOfExpr g true false PossiblyMutates h None m 
-            Some (wrap (Expr.Op (TOp.TraitCall traitInfo, [], (h' :: t), m)))
+        if minfo.IsStruct && minfo.IsInstance then 
+            match argExprs with
```
Comment (cartermp) [inline src/fsharp/MethodCalls.fs:1871]: This isn't quite the same, it's missing the `[] -> false` case.

Comment (cartermp) [reply src/fsharp/MethodCalls.fs:1871]: I also suffer from boolean logic, which is also why I think it'd be better if the original `[] -> false` case was restored

Comment (forki) [reply src/fsharp/MethodCalls.fs:1871]: Or maybe more correct: one of the two tests for [] is still in place

Comment (KevinRansom) [reply src/fsharp/MethodCalls.fs:1871]: It is, I see it now, I was always hopeless at where's waldo.

Comment (cartermp) [reply src/fsharp/MethodCalls.fs:1871]: aha, I think I see it now too. The original code was so hard to read

**Pull request merged**